src/Raven.Client/Documents/Session/DocumentQuery.cs

```diff
@@ -1176,7 +1176,12 @@ internal DocumentQuery<TResult> CreateDocumentQueryInternal<TResult>(QueryData q
                 newFieldsToFetch = FieldsToFetchToken.Create(fields, queryData.Projections.ToArray(), queryData.IsCustomFunction, sourceAlias);
             }
             else
+            {
                 newFieldsToFetch = null;
+                if (FieldsToFetchToken != null)
+                    queryData = new QueryData(FieldsToFetchToken.FieldsToFetch, FieldsToFetchToken.Projections, FromToken.Alias, DeclareTokens, LoadTokens,
```

Comment:
below we are creating new lists/linkedLists, I'm pretty sure that this is important not to pass same instance of list down there. You probably need to do the same here.

Also I see that we are doing `queryData?.ProjectionBehavior ?? ProjectionBehavior`, is this handled? I do not see that you are copying it here? Can you check if this handled properly?